src/wrapper/wrap_silo.cpp

```diff
@@ -1192,20 +1221,17 @@ namespace
         bool first = true;
         int vlength = 0;
 
-        PYTHON_FOREACH(var_py, vars_py)
+        for(py::object py_var: py_vars)
         {
-          numpy_vector<T> v = extract<numpy_vector<T> >(var_py);
+          py::array_t<T> v = py_var.cast<py::array_t<T>>();
           if (first)
           {
             vlength = v.size();
             first = false;
           }
           else if (vlength != int(v.size()))
-            PYTHON_ERROR(ValueError,
-                boost::str(boost::format(
-                    "field components of '%s' need to have matching lengths")
-                  % vname).c_str());
-          vars.push_back((float *) v.data().data());
+            PYTHON_ERROR(ValueError, "field components need to have matching lengths");
```

Comment:
Is there a nice way to do this string formatting? I could only think of `sprintf`

Reply:
https://stackoverflow.com/a/26221725

Reply:
Thanks! Updated in 6233489.